src/server.ts

```diff
@@ -83,44 +82,52 @@ export function createRemixHeaders(
 }
 
 export function createRemixRequest(
-  request: FastifyRequest,
+  fastifyReq: FastifyRequest,
   abortController?: AbortController
 ): NodeRequest {
-  let origin = `${request.protocol}://${request.hostname}`;
-  let url = new URL(request.url, origin);
+  let origin = `${fastifyReq.protocol}://${fastifyReq.hostname}`;
+  let url = new URL(fastifyReq.url, origin);
 
   let init: NodeRequestInit = {
-    method: request.method,
-    headers: createRemixHeaders(request.headers),
+    method: fastifyReq.method,
+    headers: createRemixHeaders(fastifyReq.headers),
     signal: abortController?.signal,
     abortController,
   };
 
-  if (request.method !== "GET" && request.method !== "HEAD") {
-    init.body = request.body as any;
+  if (fastifyReq.method !== "GET" && fastifyReq.method !== "HEAD") {
+    init.body = fastifyReq.body as NodeRequestInit["body"] | undefined;
   }
 
   return new NodeRequest(url.href, init);
 }
 
 function sendRemixResponse(
-  reply: FastifyReply,
-  response: NodeResponse,
+  fastifyRes: FastifyReply,
+  nodeRes: NodeResponse,
   abortController: AbortController
 ): void {
-  reply.code(response.status);
+  fastifyRes.code(nodeRes.status);
 
-  reply.headers(response.headers.raw());
+  for (const [key, values] of Object.entries(nodeRes.headers.raw())) {
```

Comment:
this will set correct headers for fastify, previously it was breaking plugins that expected string header but got array.